app/controllers/users_controller.rb

```diff
@@ -1,4 +1,132 @@
+require 'will_paginate/array'
 class UsersController < ApplicationController
+
+  before_action :require_login, only: [:twitter_search, :twitter_search_user, :vimeo_search, :vimeo_search_user, :instagram_search, :instagram_search_user, :twitter_subscribe, :vimeo_subscribe, :instagram_subscribe]
+
+  def twitter
+    Seemore::Application.config.twitter
+  end
+  #
+  # def vimeo
+  #   Seemore::Application.config.vimeo
+  # end
+
   def show
+    # Updates existing subscriptions with new content
+    Subscription.update_stories
+
+    @stories = []
+    if !@current_user
+      @stories = UsersHelper.default_content
+    else
+      @stories = UsersHelper.user_content(@current_user)
+    end
+    @stories = @stories.sort_by! { |story| story[:post_time] }.reverse!
+    @stories = @stories.paginate(:page => params[:page], :per_page => 15)
+  end
+
+  def twitter_search
+    search_term = params[:search]
+    @search_results = twitter.user_search(search_term).take(10)
+  end
+
+  def instagram_search
+    client = Instagram.client(:access_token => ENV["INSTAGRAM_ACCESS_TOKEN"])
+    search_term = params[:search]
+    @search_results = client.user_search(search_term)
+  end
+
+  def twitter_search_user
+    subscriptions = @current_user.subscriptions
+    @user_name = params[:id]
+    @user_tweets = twitter.user_timeline(@user_name)
+    uid = @user_tweets[0].user.id
+    subscription = Subscription.find(uid, "twitter")
+    if !subscriptions.include? subscription
+      @button = true
+    end
+  end
+
+  def instagram_search_user
+    subscriptions = @current_user.subscriptions
+    @user_name = params[:id]
+    @results = HTTParty.get("https://www.instagram.com/#{@user_name}/media/")
```

Comment:
Extracting out the API call into it's own method would save time later if Instagram changes their API. It would reduce the method's dependency on Instagram's API staying exactly the same. You could group your third-party API calls in one place, and make changes there, rather than hunting down each method that makes an API call.